vendor/engines/secure_rooms/app/services/secure_rooms/access_manager.rb

```diff
@@ -0,0 +1,48 @@
+module SecureRooms
+
+  class AccessManager
+
+    attr_reader :verdict
+
+    def initialize(user, card_reader, requested_account_id)
+      @user = user
+      @card_reader = card_reader
+      @requested_account_id = requested_account_id
+    end
+
+    def process
+      @verdict = create_verdict
+      event = create_event(verdict)
+    end
+
+    # TODO: Extract
+    def create_verdict
+      SecureRooms::CheckAccess.new.authorize(
+        @user,
+        @card_reader,
+        accounts: accounts,
+        selected_account: selected_account,
+      )
+    end
+
+    # TODO: Extract
+    def create_event(verdict)
+      Event.create!(
+        occurred_at: Time.current,
+        card_reader: @card_reader,
+        user: @user,
+        outcome: verdict,
```

Comment:
What actually gets written to the DB? I don't think we have a `to_s` on `Verdict` yet, so I'd be surprised if it writes the `result_code`.